### PR TITLE
doc update to use new `dart` tool in `pub upgrade` and `pub cache`

### PR DESCRIPTION
**Pull request by KevinRohn**
Updated documentation for `src/tools/pub/cmd/pub-upgrade.md` to use the new dart tool.

I also updated the `src/tools/pub/cmd/_pub-outdated-output.html` file and changed the terminal preview (html file) from `pub outdated` to `dart pub outdated`.

Tested the new `dart` tool commands and updated the doc with the new syntax.

To keep things simple I updated the example code `pub cache add barback --version "<=0.8.0 <0.110"` to `dart pub cache add http --version "0.12.2"`. The `barback` package is already archived and if someone reads it, I think a package name like `http` is more clear to understand.

See: #2617

### CODE REVIEW

File: src/tools/pub/cmd/pub-upgrade.md [hunk 1]
```diff
@@ -1,56 +1,56 @@
 ---
-title: pub upgrade
-description: Use pub upgrade to get the latest versions of all dependencies used by your Dart app.
+title: dart pub upgrade
+description: Use dart pub upgrade to get the latest versions of all dependencies used by your Dart app.
 ---
 
 _Upgrade_ is one of the commands of the [pub tool](/tools/pub/cmd).
```
Comment (kwalrath) [inline src/tools/pub/cmd/pub-upgrade.md:6]: We should probably say something about `pub upgrade` vs. `dart pub upgrade`, using some standard wording. (Also see #2826.)

File: src/tools/pub/cmd/pub-cache.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
 ---
-title: pub cache
-description: Use pub cache to manage your system cache.
+title: dart pub cache
+description: Use dart pub cache to manage your system cache.
 toc: false
 ---
 
 _Cache_ is one of the commands of the [pub tool](/tools/pub/cmd).
```
Comment (kwalrath) [inline src/tools/pub/cmd/pub-cache.md:8]: Same comment as for pub-upgrade: we should explicitly talk about `pub` vs. `dart pub`.

File: src/tools/pub/cmd/_pub-outdated-output.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
 transitive dev_dependencies: all up-to-date
 
 1 upgradable dependency is locked (in pubspec.lock) to an older version.
-To update it, use `pub upgrade`.
+To update it, use `dart pub upgrade`.
```
Comment (kwalrath) [inline src/tools/pub/cmd/_pub-outdated-output.html:16]: Unfortunately, the output still says `pub upgrade`, not `dart pub upgrade` (at least in 2.10.4 stable, and we document the stable version). I've filed https://github.com/dart-lang/sdk/issues/44612.

/cc @mit-mit

**Review by kwalrath — COMMENTED** [review]
LGTM, aside from the command output.

Staged:
https://kw-www-dartlang-1.firebaseapp.com/tools/pub/cmd/pub-cache
https://kw-www-dartlang-1.firebaseapp.com/tools/pub/cmd/pub-upgrade
https://kw-www-dartlang-1.firebaseapp.com/tools/pub/cmd/pub-outdated (which is where _pub-outdated-output.html is included)

**Comment by kwalrath** [comment]
@mit-mit or @johnpryan, any opinion on changing barback to http?

File: src/tools/pub/cmd/pub-upgrade.md [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
 versions of all the dependencies listed in the
 [`pubspec.yaml`](/tools/pub/pubspec) file in the current working
 directory, as well as their [transitive
 dependencies](/tools/pub/glossary#transitive-dependency).
 For example:
 
 ```terminal
-$ pub upgrade
+$ dart pub upgrade
 Dependencies upgraded!
```
[the remark this answers is not in the document]
Comment (KevinRohn) [reply src/tools/pub/cmd/pub-upgrade.md:29]: Let me check this later. I think the output was exactly the same as in the code. Maybe it depends on the OS? I tested it with windows. I will try it also with my Mac.

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/tools/pub/cmd/pub-upgrade.md:29]: Just to document my process... Before running these commands, I used `dart create`, changed to the new directory, and then changed the pubspec.lock file to have one version earlier of meta & pedantic.

File: src/tools/pub/cmd/_pub-outdated-output.html (hunk 3 above)
Comment (mit-mit) [reply src/tools/pub/cmd/_pub-outdated-output.html:16]: Fixing in https://github.com/dart-lang/pub/pull/2823 (note: after that is merged, it'll take a bit of time to roll the change in the the SDK. But I think I'd be fine to update docs as soon as the PR is merged).

File: src/tools/pub/cmd/pub-upgrade.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (KevinRohn) [reply src/tools/pub/cmd/pub-upgrade.md:29]: @kwalrath  Sorry I'm quite busy this days. I will try to do it on the weekend.

**Comment by johnpryan** [comment]
Sorry for the late response - yes `http` is better.

**Review by kwalrath — APPROVED** [review]
I'm going to accept this change. We can always change the output (if necessary) later. Thank you, @KevinRohn!

**Pull request merged**